src/ui/components/Timeline/Capsule.tsx

```diff
@@ -55,13 +59,82 @@ function LoadingState({
   const onMouseLeave = () => setShowLoadingProgress(false);
 
   return (
-    <div className={styles.LoadingState} onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
-      {loadingStatusSlow ? (
+    <>
+      <div className={styles.LoadingState} onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
+        {loadingStatusSlow ? <SlowLoadingIcon /> : <RadialProgress progress={progress} />}
+        <div className={styles.LoadingText}>{Math.round(progress)}%</div>
+      </div>
+    </>
+  );
+}
+
+function SlowLoadingIcon() {
+  const ref = useRef<HTMLDivElement>(null);
+  const [showSlowLoadingPopout, setShowSlowLoadingPopout] = useState<boolean>(false);
+
+  useEffect(() => {
+    const div = ref.current;
+    if (!div) {
+      return;
+    }
+
+    let timeoutID: NodeJS.Timeout | null = null;
+
+    const showAfterTimeout = () => {
+      timeoutID = setTimeout(() => {
+        timeoutID = null;
+
+        setShowSlowLoadingPopout(true);
+      }, SHOW_SLOW_LOADING_POPOUT_AFTER_DELAY);
+    };
+
+    const clearPendingTimeout = () => {
+      if (timeoutID !== null) {
+        clearTimeout(timeoutID);
+      }
+    };
+
+    div.addEventListener("click", clearPendingTimeout);
```

Comment:
I'm having a hard time picturing this interaction. Which element is this click handler on and which part of the interaction does it handle?

Reply:
Basically I'm setting a hover timeout (as per Jon's request) that we shouldn't show the popout _immediately_ but we should "wait a sec". So I'm just cleaning up the timeout if the user explicitly clicks to show it (or if the thing unmounts)

Reply:
Ah, gotchya, in that case, what do you think about putting this in the `onClick` handler on line 112 where we show the popout explicitly??

Reply:
I think that makes sense, except that we'd then need to store the timeout ID in a ref so it could be referenced by the React click handler (since we're inside of an effect now).

TBH it's just a little optimization. Setting the value to true when it's already true will bailout anyway so if it's this line of code is a snag, I can just remove it entirely :)

Reply:
Nah, that makes sense. I definitely would not want to add a ref just to move a line of code around for readability, since that would probably add more cognitive load than it would relieve. All good! 🚢

Reply:
I appreciate that 👍🏼 

I'm going to walk back my claim about it being an unneccessary optimization. It's true that React would be able to bailout without rendering if we were setting an already-true value to true, _but_ we'd have a potential edge case bug if a user hovered, clicked (to show) and then clicked again (to hide) before the timer fired– we'd end up re-showing the popout.

So I'm going to just leave it in place 😁